### ISSUE TITLE
dmonsfree: 1 removed doesn't match 2 pending on Dlvl:1  (you polyd as a spellcaster)

### ISSUE THREAD

**Issue by elunna** [elunna/hackem#134]
```
(rr) p toplines
$25 = "You kill the koala!  You are caught in the torrent of acid!  You kill the koala!", '\000' <repeats 219 times>


#0  __pthread_kill_implementation (no_tid=0, signo=6, threadid=140225544640448) at ./nptl/pthread_kill.c:44
#1  __pthread_kill_internal (signo=6, threadid=140225544640448) at ./nptl/pthread_kill.c:78
#2  __GI___pthread_kill (threadid=140225544640448, signo=signo@entry=6) at ./nptl/pthread_kill.c:89
#3  0x00007f88cd81a476 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#4  0x00007f88cd8007f3 in __GI_abort () at ./stdlib/abort.c:79
#5  0x000055acb78a5cce in NH_abort () at end.c:236
#6  0x000055acb78acbc7 in panic (str=0x55acb83b4280 "%s") at end.c:802
#7  0x000055acb7e0dd5b in impossible (s=0x55acb836be60 "dmonsfree: %d removed doesn't match %d pending on %s")
    at pline.c:518
#8  0x000055acb7c1e403 in dmonsfree () at mon.c:3022
#9  0x000055acb7c0a9fc in movemon () at mon.c:1579
#10 0x000055acb75b523e in moveloop (resuming=0 '\000') at allmain.c:244
#11 0x000055acb823fa46 in main (argc=0, argv=0x7ffebf1a0cf8) at ../sys/unix/unixmain.c:353


full backtrace
...
#6  0x000055acb78acbc7 in panic (str=0x55acb83b4280 "%s") at end.c:802
        the_args = {{gp_offset = 16, fp_offset = 48, overflow_arg_area = 0x7ffebf1a0280, 
            reg_save_area = 0x7ffebf1a01b0}}
#7  0x000055acb7e0dd5b in impossible (s=0x55acb836be60 "dmonsfree: %d removed doesn't match %d pending on %s")
    at pline.c:518
        pbuf = "dmonsfree: 1 removed doesn't match 2 pending on Dlvl:1  \000\000\000\000\000\000\000\000\001\000\000\000\001", '\000' <repeats 19 times>, "\364Д\267\254U\000\000\000\262\020\000\020a\000\000\a\004\032\277\v\177\000\000\220\003\032\277\001\177\000\000\000\262\020\000\020a\000\000\240\003\032\277\376\177\000\000\256ы\267\254U\000\000\000\000\000\000\003\000\000\000\000\261\377\355[\211\345\376\300\003\032\277\376\177\000\000\000\262\020\000\020a\000\000\02--
        \000\000\003\000\000\000\000\261\377\355[\211\345\376\300\003\032\277\376\177\000\000\000\262\020\000\020a\000\000\02--Type <RET> for more, q to quit, c to continue without paging--
0\004\032\277\376\177\000\000\202@\343\327\377\017\000\000\220\005\032\277\376\177\000\000\340\236"...
        the_args = {{gp_offset = 32, fp_offset = 48, overflow_arg_area = 0x7ffebf1a0970, 
            reg_save_area = 0x7ffebf1a0890}}
#8  0x000055acb7c1e403 in dmonsfree () at mon.c:3022
        mtmp = 0x61100010bfc0
        freetmp = 0x61100010bfc0
        count = 1
        buf = "Dlvl:1  \000Q\221\267?\000\000\000\000\n\032\277\376\177\000\000\337ye\267\254U\000\000\320\n\032\277\376\177\000\000\000\000\000\000\004\000\000\000T\242d\270\254U\000\000j\354e\270\254U\000\000 \n\032\277\376\177\000\000K\366l\267\254U\000\000P\n\032\277\376\177\000\000\017\000\000\000B\000\000\000\022\000\000\000\000\000\000\000\071\001\000\000\000\000\000\031P\n\032\277\376\177\000\000\000\261\377\355[\211\345\376"
#9  0x000055acb7c0a9fc in movemon () at mon.c:1579
        mtmp = 0x0
        nmtmp = 0x0
        somebody_can_move = 0 '\000'
#10 0x000055acb75b523e in moveloop (resuming=0 '\000') at allmain.c:244
        moveamt = 21
        wtcap = 0
        change = 0
        monscanmove = 0 '\000'
        elf_regen = 1 '\001'
        orc_regen = 1 '\001'
        vamp_regen = 1 '\001'
#11 0x000055acb823fa46 in main (argc=0, argv=0x7ffebf1a0cf8) at ../sys/unix/unixmain.c:353
        fd = -1
        dir = 0x0
        exact_username = 0 '\000'
        resuming = 0 '\000'
        plsel_once = 1 '\001'
```

**Comment by elunna** [comment]
I was able to reproduce this in EvilHack as well:
```
 Program in disorder!  (Saving and reloading may fix this problem.)
 |.....% dmonsfree: 1 removed doesn't match 2 pending on Dlvl:2
 |....$. You kill the rothe!  You are caught in the ball of cold!  You kill the rothe!        |...... You blast the rothe with ice!  The rothe is caught in the ball of cold!
 ------- You touch the rothe.  The rothe is covered in frost!  You cast a spell!
         Create what kind of monster? [type the name or symbol] rothe

```

**Comment by k21971** [comment]
See https://github.com/k21971/EvilHack/commit/c110d5ea9ac2784853eaae38924148cce8ae3df1

**Comment by elunna** [comment]
Fixed in fdaafb93a
